packages/coverage-istanbul/src/provider.ts

```diff
@@ -175,6 +181,36 @@ export class IstanbulCoverageProvider implements CoverageProvider {
       }
     }
   }
+
+  async includeUntestedFiles(coverageMap: CoverageMap) {
+    const files = await this.testExclude.glob(this.ctx.config.root)
+
+    for (const file of files) {
```

Comment:
Maybe we should do `transformRequest` inside `Promise.all`?

Reply:
Sure. Should we still set some maximum limit of parallel `transformRequest` calls - so that we don't end up transforming 1000 files inside a single `Promise.all`. Maybe loop the `files` in sets of 20-25?

Reply:
Now that you say this, I guess we should 🤔 We already use `p-limit` library for `.concurrent`, so you can reuse it here. But I think we can be liberal with the maximum number, tbh - maybe base it on cores/available memory?

We already do `Promise.all` on codebase here with our limit tho:

https://github.com/vitest-dev/vitest/blob/5c064944126c36f3e1df0ecd5289da3b67b4353b/packages/vitest/src/node/core.ts#L184

Reply:
Now using `p-limit` and limiting parallel transforms with the count of threads (minus one so that main thread is not blocked). I tested this setup with +100 generated uncovered files, with artificial timeout. Batching seems to work nicely. 

One thing to pay attention here @sheremet-va. I added `p-limit` as devDependency - this seems wrong. In `packages/vitest` it is declared as devDependency. Shouldn't it be regular dependency since it is used in runtime by consumers? 

https://github.com/vitest-dev/vitest/blob/03f7450957de6d18826040c7c5a1ea98712b251c/packages/vitest/package.json#L133

Reply:
> Shouldn't it be regular dependency since it is used in runtime by consumers?

dev dependencies are inlined, when Vitest is bundled, so no, it shouldn't.